lib/AST/Bridging/DeclAttributeBridging.cpp

```diff
@@ -487,6 +487,39 @@ BridgedSwiftNativeObjCRuntimeBaseAttr_createParsed(BridgedASTContext cContext,
       SwiftNativeObjCRuntimeBaseAttr(name, atLoc, range, /*Implicit=*/false);
 }
 
+BridgedWarnAttr
+BridgedWarnAttr_createParsed(BridgedASTContext cContext,
+                             SourceLoc atLoc,
+                             SourceRange range,
+                             Identifier diagGroupName,
+                             WarningGroupBehavior behavior,
+                             BridgedStringRef reason) {
+  ASTContext &context = cContext.unbridged();
+  auto diagGroupID = getDiagGroupIDByName(diagGroupName.str());
+  
+  WarningGroupBehavior attrBehavior;
+  switch (behavior) {
+    case None:
+    case AsWarning:
+      attrBehavior = WarningGroupBehavior::AsWarning;
+      break;
+    case AsError:
+      attrBehavior = WarningGroupBehavior::AsError;
+      break;
+    case Ignored:
+      attrBehavior = WarningGroupBehavior::Ignored;
+      break;
+  }
+  
+  std::optional<StringRef> reasonText = std::nullopt;
+  if (!reason.getIsEmpty())
+    reasonText = reason.unbridged();
```

Comment:
Don't we need to copy this string into the ASTContext?

Reply:
Good catch, thank you.